### PR TITLE
Fix structure and execution of type tests

### PR DESCRIPTION
**Pull request by martint**
Do not use inheritance/overriding for code reuse and specialization, as it makes it harder to reason about what's being tested.

Also, annotate the tests directly so that they are picked up by JUnit

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by hashhar — COMMENTED** [review]
how do we plan to deal with evolving tests? e.g. if you add a new test `testRange` then how do you identify which all test classes it should go into? Similarly how do you ensure there isn't a drift in the test method definition across classes.

**Review by findinpath — APPROVED** [review]

**Comment by martint** [comment]
Each type needs tests according to its semantics. If you add new behavior to a type that affects how getRange() works, the corresponding tests for that type need to be adjusted. That's no different than how we implement tests for timestamp, timestamp w/ tz, time, time w/ tz, etc. in io.trino.operator.scalar.timestamp.TestTimestamp, io.trino.operator.scalar.timestamp.TestTime, and friends.

File: core/trino-main/src/test/java/io/trino/type/AbstractTestType.java [hunk 1]
```diff
@@ -203,41 +202,6 @@ public void testBlock()
...
-        assertThat(type.getRange())
-                .isEmpty();
-    }
-
-    @Test
-    public void testPreviousValue()
-    {
-        Object sampleValue = getSampleValue();
-        if (!type.isOrderable()) {
-            assertThatThrownBy(() -> type.getPreviousValue(sampleValue))
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/type/AbstractTestType.java:218]: Would it make sense to at least cc the author of this code on the pr that fixes it?

Comment (martint) [reply core/trino-main/src/test/java/io/trino/type/AbstractTestType.java:218]: It was mentioned in this thread, which you were already part of: https://github.com/trinodb/trino/pull/18588#discussion_r1288304618

**Pull request merged**